learn/getting-started/writing-a-graphql-api-with-ballerina.md

```diff
@@ -0,0 +1,397 @@
+    recovered: Decimal
+    cases: Decimal
+    active: Decimal
+    deaths: Decimal
+}
+```
+
+> Note: The `Decimal` type used here is coming from the `decimal` type in Ballerina language. We are using this type
+> to store the case counts. More about this later.
+
+Now that we have the design, we can create the project.
+
+## Create a Ballerina Project
+
+1. Open a terminal and move to the directory where you want to create the Ballerina project.
+
+   On Linux or Mac:
+    ```shell
+    cd your/parent/directory/path
+    ```
+
+   On Windows:
+    ```windows
+    cd your\parent\directory\path
+    ```
+2. Using the `bal` command, create a new Ballerina project.
+    ```shell
+    bal new covid19
+   ```
+This will create a new Ballerina project inside a directory named `covid19`.
+
+## Create a Datasource for the Project
+
+Before writing the GraphQL service, we are going to create a data source for out project. This will mimic a database
+where we store the data for our service. Use the following command to create a new module inside the Ballerina project.
+
+```shell
+bal add datasource
+```
+
+This will create a new directory named `modules/datasource`, which will be the Ballerina module acting as the data
+source of the GraphQL service.
+
+Now, let's add the data needed for the GraphQL service inside this module.
+
+First, we have to define our data types. We can use Ballerina records for this. Let's first define a Ballerina record to
+store an entry in the database. Although the Covid-19 database has many fields, for this guide we only use the `isoCode`
+, `continent`, `location`, `date`, `totalCases`, and `newCases`. Let's create a record with these fields.
+
+To do so, let's create a file `types.bal` inside the `modules/datasource` directory and add the following record
+definition there.
+
+```ballerina
+public type CovidEntry record {|
+    readonly string isoCode;
+    string country;
+    decimal cases?;
+    decimal deaths?;
+    decimal recovered?;
+    decimal active?;
+|};
+```
+
+Now we need a table to store the sample data. For this tutorial, we are going to use an in-memory table with just three
+entries. Let's define the table inside the auto-created `datasource.bal` file as follows:
+
+```ballerina
+isolated table<CovidEntry> key(isoCode) covidEntriesTable = table [
+    {isoCode: "AFG", country: "Afghanistan", cases: 10000, deaths: 500, recovered: 1500, active: 8000},
+    {isoCode: "SL", country: "Sri Lanka", cases: 20000, deaths: 1000, recovered: 4000, active: 15000},
+    {isoCode: "US", country: "United States", cases: 40000, deaths: 5000, recovered: 15000, active: 20000},
+]
+```
+
+Then we can define functions to mock the database operations. Although we use an in-memory table, we can use any data
+source and use the same functions to access them, so that the GraphQL service will not have any impact when the
+datasource is changed.
+
+First, the following function will return all the entries from the table.
```

Comment:
```suggestion
First, the following function will return all the entries from the table.
```
Consider adding numbered bullet points always if a sequence of steps/instructions is presented.